include/ecdaa/prng.h

```diff
@@ -0,0 +1,110 @@
+/******************************************************************************
+ *
+ * Copyright 2017 Xaptum, Inc.
+ * 
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ * 
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License
+ *
+ *****************************************************************************/
+
+#ifndef ECDAA_RANDOM_H
+#define ECDAA_RANDOM_H
+#pragma once
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <amcl/randapi.h>
+
+#include <stddef.h>
+
+#define AMCL_SEED_SIZE 128  // As recommended in AMCL's "rand.c"
+
+enum ecdaa_prng_initialized {
+    ECDAA_PRNG_INITIALIZED_NO = 0,
+    ECDAA_PRNG_INITIALIZED_YES
+};
+
+/*
+ * Wrapper around AMCL's pseudo-random-number-generator.
+ *
+ * For the safe operation of this library,
+ * all `ecdaa_prng`s MUST first be properly seeded
+ * via a call to `ecdaa_prng_init`
+ * of `ecdaa_prng_init_custom` before first use.
+ */
+struct ecdaa_prng {
+    enum ecdaa_prng_initialized initialized;
+    csprng impl;
+};
+
+/*
+ * Accessor for the underlying AMCL csprng in a `ecdaa_prng`.
+ *
+ * Checks that `ecdaa_prng` was initialized before returning.
+ * If not, calls `abort()`.
+ */
+csprng *get_csprng(struct ecdaa_prng *prng);
+
+/*
+ * Securely-clears memory used by a `ecdaa_prng`.
+ *
+ * To be used when finished with a `ecdaa_prng`.
+ *
+ * Does not do any heap deallocation.
+ */
+void ecdaa_prng_clean(struct ecdaa_prng *prng);
```

Comment:
Rename to `ecdaa_prng_free()`to match C convention?

Reply:
After addressing this, I say merge it.  The public API seems to be good.

The nature of the inner usage can evolve (what the `acml-conventions` should be will become more clear as support for other curves are added.  Whether `get_csrng` should be `prng_get_csrng` or `ecdaa_prng_get_csrng` will become more clear when naming conventions for internal code are decided. For now there are bigger fish to fry.)

Reply:
I avoided `free` here to keep from insinuating memory is being freed. But, maybe making people think that would be good (more people have been burned by memory leaks than by side-channel attacks, at least that they know of).

Reply:
Yea, bigger fish indeed.

I spoke with Michael Scott from AMCL a fair amount today, getting advice on using their csprng. Definitely feel more confident in their implementation and our usage of it now. So, I'm go for this PR, other than your input on `free` vs `clean`.